File: nexus-realworld/apps/server/prisma/seed.ts
import { context } from "~/context";
import bcrypt from "bcryptjs";

const { db: prisma } = context;

async function seed() {
  await prisma.message.deleteMany();
  await prisma.article.deleteMany();
  await prisma.user.deleteMany();

  for (let i = 0; i < 10; i++) {
    const user = await prisma.user.create({
      data: {
        name: `user${i}`,
        email: `user${i}@example.com`,
        passwordHash: bcrypt.hashSync("password"),
      },
    });
    for (let j = 0; j < 100; j++) {
      await prisma.article.create({
        data: {
          title: `user${i}-title${j}`,
          description: `user${i}-description${j}`,
          body: `user${i}-body${j}`,
          authorId: user.id,
        },
      });
    }
  }

  const [user1, user2] = await prisma.user.findMany();
  for (let i = 0; i < 10; i++) {
    await prisma.message.create({
      data: {
        text: `${user1.name} -> ${user2.name}`,
        fromUserId: user1.id,
        toUserId: user2.id,
      },
    });
  }
}

seed();

File: go-gqlgen-sqlc/scripts/seed.go
//go:build seed
// +build seed

package main

import (
	"context"
	"database/sql"
	"fmt"
	"go-gqlgen-sqlc/db"
	"go-gqlgen-sqlc/graphql/scalar"
	"log"
	"os"

	"github.com/google/uuid"
	"github.com/joho/godotenv"
	"golang.org/x/crypto/bcrypt"
)

func main() {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env file")
	}
	database, err := sql.Open("postgres", os.Getenv("DATABASE_URL"))
	if err != nil {
		log.Fatal(err)
	}

	q := db.New(database)
	ctx := context.Background()

	for u := 0; u < 10; u++ {
		user, err := q.CreateUser(ctx, &db.CreateUserParams{
			ID:           scalar.UUID{UUID: uuid.New()},
			Name:         fmt.Sprintf("user%d", u),
			Email:        fmt.Sprintf("user%d@example.com", u),
			PasswordHash: hash("password"),
		})
		if err != nil {
			log.Fatal(err)
		}

		for p := 0; p < 100; p++ {
			_, err = q.CreatePost(ctx, &db.CreatePostParams{
				ID:     scalar.UUID{UUID: uuid.New()},
				UserID: user.ID,
				Title:  fmt.Sprintf("user%d-title%d", u, p),
				Body:   fmt.Sprintf("user%d-body%d", u, p),
			})
			if err != nil {
				log.Fatal(err)
			}
		}
	}
}

func hash(password string) string {
	h, _ := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
	return string(h)
}
